packages/react-aria-components/src/Menu.tsx

```diff
@@ -112,29 +161,35 @@ function MenuInner<T extends object>({props, collection, menuRef: ref}: MenuInne
           return <Separator {...item.props} />;
         case 'item':
           return <MenuItemInner item={item} />;
+        case 'submenutrigger':
+          return <SubmenuTriggerInner item={item} parentMenuRef={ref} />;
         default:
           throw new Error('Unsupported node type in Menu: ' + item.type);
       }
     }
   });
 
   return (
-    <div
-      {...filterDOMProps(props)}
-      {...menuProps}
-      ref={ref}
-      slot={props.slot || undefined}
-      onScroll={props.onScroll}
-      style={props.style}
-      className={props.className ?? 'react-aria-Menu'}>
-      <Provider
-        values={[
-          [MenuStateContext, state],
-          [SeparatorContext, {elementType: 'div'}]
-        ]}>
-        {children}
-      </Provider>
-    </div>
+    <FocusScope>
+      <div
+        {...filterDOMProps(props)}
+        {...menuProps}
+        ref={ref}
+        slot={props.slot || undefined}
+        onScroll={props.onScroll}
+        style={props.style}
+        className={props.className ?? 'react-aria-Menu'}>
+        <Provider
+          values={[
+            [MenuStateContext, state],
+            [SeparatorContext, {elementType: 'div'}],
+            [SubmenuContext, {popoverContainerRef}]
+          ]}>
+          {children}
+        </Provider>
+      </div>
+      {rootMenuTriggerState?.isOpen && <div ref={popoverContainerRef} style={{width: '100vw', position: 'absolute', top: -5, ...leftOffset}} /> }
```

Comment:
This `-5` for top was added to Spectrum to help with some of the offsetting/positioning of the submenu, but for RAC do we want to support the end user specifying some kind of offset at a higher level? I think the Popover allows for specifying the offsets but it might still be bounded by the positioning of this div... Might be hard, not sure

Reply:
Removed the -5. Users can set the popover offset, but not sure if they would need to adjust the container's.